src/migration.ts

```diff
@@ -49,19 +50,71 @@ export type CreateOptions = {
 
 const SEPARATOR = '_';
 
+function localeCompareStringsNumerically(a: string, b: string): number {
+  return a.localeCompare(b, undefined, {
+    usage: 'sort',
+    numeric: true,
+    sensitivity: 'variant',
+    ignorePunctuation: true,
+  });
+}
+
+function compareFileNamesByTimestamp(
+  a: string,
+  b: string,
+  logger?: Logger
+): number {
+  const aTimestamp = getTimestamp(a, logger);
+  const bTimestamp = getTimestamp(b, logger);
+
+  return aTimestamp - bTimestamp;
+}
+
+/**
+ * ! this function does not actually load the files - it only reads their names / paths from `dir`
+ * ? should the function body be adjusted to do only what can be expected from it's name
+ */
 export async function loadMigrationFiles(
-  dir: string,
-  ignorePattern?: string
+  dir: string | string[],
+  ignorePattern?: string | string[],
+  useGlob: boolean = false,
+  logger?: Logger
 ): Promise<string[]> {
```

Comment:
**polish:** when we have 3 or more parameters where many of them are even optional, we can refactor it to use an options pattern.
So in this case we have one required parameter and 3 options, which we can then refactor to

```ts
interface LoadMigrationFilesOptions {
  /**
   * ...
   */
  ignorePattern?: string | string[];
  /**
   * ...
   *
   * @default false
   */
  useGlob?: boolean;
  /**
   * ...
   */
  logger?: Logger;
}

function loadMigrationFiles(
  dir: string | string[],
  options: LoadMigrationFilesOptions = {}
): Promise<string[]> {
  const { ignorePattern, useGlob = false, logger } = options;
```

_we need to check if this function is externally exported and used by intention from outside_

Reply:
totally agree. I only didn't do this so that this PR doesn't introduce a breaking change

Reply:
it is not exported from the `index.js` file - however, since the package specifies wildcard exports, a consumer could do 

```js
import { loadMigrationFiles } from 'node-pg-migrate/migration';
```

Reply:
> it is not exported from the `index.js` file - however, since the package specifies wildcard exports, a consumer could do
> 
> ```js
> import { loadMigrationFiles } from 'node-pg-migrate/migration';
> ```

this is what I wanted to check 😄